test/integration/puppetlabs/services/jruby/jruby_puppet_pool_int_test.clj

```diff
@@ -384,6 +386,90 @@
+                          (set (map :metric-name url-metrics-data))))
+                   (is (= 2 (:count (first url-metrics-data))))
+                   (is (= #{(add-metric-ns "with-url-and-method.http://localhost:8080/hello.GET")
+                            (add-metric-ns "with-url-and-method.http://localhost:8080/hello.POST")}
+                          (set (map :metric-name (:url-and-method metrics-data)))))))
+               (testing "no metric-id metrics are registered"
+                 (is (= [] (:metric-id (metrics/get-client-metrics-data metric-registry))))))
+             (testing (str "metric registry has metric-id metrics after http requests have been made"
+                           " with metric-id specified")
+               (testing "GET request"
+                 (.runScriptlet container
+                                "$c.get('/hello', {}, {:metric_id => ['foo', 'get']})")
+                 (let [metric-id-data (:metric-id (metrics/get-client-metrics-data metric-registry))]
+                   (is (= 2 (count metric-id-data)))
+                   (is (= #{(add-metric-ns "with-metric-id.foo")
+                            (add-metric-ns "with-metric-id.foo.get")}
+                          (set (map :metric-name metric-id-data)))))
+                 (is (= 1 (:count (first (metrics/get-client-metrics-data-by-metric-id
+                                          metric-registry ["foo"])))))
+                 (is (= 1 (:count (first (metrics/get-client-metrics-data-by-metric-id
+                                          metric-registry ["foo" "get"]))))))
+               (testing "POST request"
+                 (.runScriptlet container
+                                "$c.post('/hello', 'body', {}, {:metric_id => ['foo', 'post']})")
+                 (let [metric-id-data (:metric-id (metrics/get-client-metrics-data metric-registry))]
+                   ;; there should now be two requests with the 'foo' metric id, one with 'foo.get',
+                   ;; and one with 'foo.post'
+                   (is (= 3 (count metric-id-data)))
+                   (is (= #{(add-metric-ns "with-metric-id.foo")
+                            (add-metric-ns "with-metric-id.foo.get")
+                            (add-metric-ns "with-metric-id.foo.post")}
+                          (set (map :metric-name metric-id-data)))))
+                 (is (= 2 (:count (first (metrics/get-client-metrics-data-by-metric-id
+                                          metric-registry ["foo"])))))
+                 (is (= 1 (:count (first (metrics/get-client-metrics-data-by-metric-id
+                                          metric-registry ["foo" "get"])))))
+                 (is (= 1 (:count (first (metrics/get-client-metrics-data-by-metric-id
+                                          metric-registry ["foo" "post"]))))))))
+           (finally
+             (jruby-testutils/return-instance jruby-service jruby-instance :no-metrics-test))))))))
```

Comment:
Seems like the last argument in this call should be `:http-client-metrics-test` - although it probably doesn't really matter.